### PR TITLE
Update Node API endpoint

### PR DESCRIPTION
**Pull request by YazzyYaz**
Updates Node API Endpoint and closes https://github.com/celestiaorg/docs/issues/65

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202262341125517

### CODE REVIEW

File: docs/developers/node-api.md [hunk 1]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
...
+#### GET /header/{height}
+Returns the header of the given height.
 
 Request
 ```sh
-curl -X GET http://<ip>:26658/balance
+curl -X GET http://<ip>:26658/header/300
+```
+Response
+```sh
```
Comment (adlerjohn) [inline docs/developers/node-api.md:18]: And elsewhere.
````suggestion
```json
````

File: docs/developers/node-api.md [hunk 2]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
 
 In order to query the API, you will need to setup your node. Resources on how to do this can be found in the following [guide](https://docs.celestia.org/nodes/bridge-validator-node#deploy-the-celestia-node).
 
-### `GET`   `/balance`
+## Endpoints
+### Header endpoints
```
Comment (adlerjohn) [inline docs/developers/node-api.md:8]: Install mdlint in your IDE to automatically detect and fix these.
```suggestion
## Endpoints

### Header endpoints
```

File: docs/developers/node-api.md [hunk 3]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
...
+```
+
+### Share endpoints
+
+#### GET /namespaced_shares/{nID}
+Returns shares of the given namespace ID nID from the latest block (chain head).
```
Comment (renaynay) [inline docs/developers/node-api.md:25]:
```suggestion
Returns shares of the given namespace ID `nID` from the latest block (chain head).
```

File: docs/developers/node-api.md [hunk 4]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
...
+#### GET /namespaced_shares/{nID}/height/{height}
+Returns shares of the given namespace ID nID from the block of the given height height.
 
+Request
+```sh
+curl -X GET http://<ip>:26658/namespaced_shares/0000000000000001/height/357889
+```
 Response
-```json
+```sh
```
Comment (renaynay) [inline docs/developers/node-api.md:44]:
```suggestion
```json
```

File: docs/developers/node-api.md [hunk 5]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
...
+
+#### GET /balance
+Returns the balance of the default account address of the node.
+
+Request
+```sh
+curl -X GET http://<ip>:26658/balance
+```
+Response
+```sh
```
Comment (renaynay) [inline docs/developers/node-api.md:58]:
```suggestion
```json
```

File: docs/developers/node-api.md [hunk 6]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
...
+Returns the balance of the given address.
 
 Request
 ```sh
 curl -X GET http://<ip>:26658/balance/celes1vuw7pdcyap2u62x6dyqjdzznlg25jll6jj76d0
 ```
-
 Response
-```json
+```sh
```
Comment (renaynay) [inline docs/developers/node-api.md:70]:
```suggestion
```json
```

File: docs/developers/node-api.md [hunk 7]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
...
 ```
+Response
+```sh
```
Comment (renaynay) [inline docs/developers/node-api.md:82]:
```suggestion
```json
```

File: docs/developers/node-api.md [hunk 8]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
...
+{"txhash":"63F2F87C2602A2F93585FB23691D17E36F9EA8F47CF50083C88941548F28B588","codespace":"sdk","code":30,"raw_log":"block height: 5361, timeout height: 99: tx timeout height","logs":null,"gas_wanted":10000}
+```
+
+#### POST /submit_pfd
+Constructs, signs and submits a PayForData message to a running instance of celestia-app.
```
Comment (renaynay) [inline docs/developers/node-api.md:87]:
```suggestion
Constructs, signs and submits a PayForData message to a running instance of celestia-app. The body of the `/submit_pfd` request should contain the hex-encoded `namespace_id`, the hex-encoded `data`, and the `gas_limit` as a uint64.
```

File: docs/developers/node-api.md [hunk 9]
```diff
@@ -4,38 +4,93 @@ This is the Celestia Node API Docs reference page for being able to make API req
...
 
+Request
+```sh
+curl -X POST -d '{"namespace_id": "0000010000000100", "data": "68656c6c6f", "gas_limit": 100000}' http://<ip>:26658/submit_pfd
+```
 Response
-```json
-{"txhash":"E4B98AF73852A2FC99E1708C3C5B7CE93449A0E9381387D022FAE759F76E1F0D","codespace":"sdk","code":0,"logs":null}
+```sh
```
Comment (renaynay) [inline docs/developers/node-api.md:94]:
```suggestion
```json
```

**Comment by YazzyYaz** [comment]
@adlerjohn thanks for the suggestion, I have made a PR to add mdlint inside the package.json https://github.com/celestiaorg/docs/pull/67

I will try to integrate this also in my IDE today, my neovim package is a mess so need to clean it up a bit.

**Comment by YazzyYaz** [comment]
@adlerjohn I pushed a linted commit of this doc. Can you review so I can pull it to the `add-lint` branch after I merge?

**Comment by adlerjohn** [comment]
@YazzyYaz the formatting looks fine, but not sure about the content. Might want @renaynay for that. In the meantime, why not fix the linting issues inside of #67 and merge that into `main`. Then other branches can be rebased off of that.

**Review by renaynay — APPROVED** [review]

**Pull request merged**